### ISSUE TITLE
[BUG] jnosql-arangodb - _system access required for any database connection

### ISSUE THREAD

**Issue by idoub** [eclipse/jnosql#562]
### Which JNoSQL project the issue refers to?

JNoSQL Databases

### Bug description

For security reasons, our primary user only has access to our primary database and not the `_system` database. Unfortunately, we are unable to establish a connection to _any_ database without access to the `_system` database.

```
Response: 401, Error: 11 - not authorized to execute this request
com.arangodb.ArangoDBException: Response: 401, Error: 11 - not authorized to execute this request
	at com.arangodb.ArangoDBException.of(ArangoDBException.java:137)
	at com.arangodb.ArangoDBException.of(ArangoDBException.java:111)
	at com.arangodb.ArangoDBException.of(ArangoDBException.java:107)
	at com.arangodb.internal.net.CommunicationProtocol.execute(CommunicationProtocol.java:45)
	at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:55)
	at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:47)
	at com.arangodb.internal.ArangoDBImpl.getDatabases(ArangoDBImpl.java:96)
	at org.eclipse.jnosql.databases.arangodb.communication.ArangoDBUtil.checkDatabase(ArangoDBUtil.java:63)
	at org.eclipse.jnosql.databases.arangodb.communication.ArangoDBDocumentManagerFactory.apply(ArangoDBDocumentManagerFactory.java:33)
```

The `checkDatabase` method in `ArangoDBUtil.java` calls `getDatabases()` which requires access to the `_system` database. The `checkDatabase` method should probably use the `getAccessibleDatabases()` or `getAccessibleDatabasesFor(String user)` methods instead.

### JNoSQL Version

1.1.2

### Steps To Reproduce

1. Create a new database in Arango.
2. Create a new user that has access to the new database but no access to the _system database.
3. Try to connect to the new database with the new user.

### Expected Results

Should be able to connect to a database with a user that has no access to the `_system` database.

**Comment by otaviojava** [comment]
Thank you for reporting it, @idoub. I am working on it.

**Comment by otaviojava** [comment]
I've released version 1.1.3 with this change; please let me know if it fixed your issue.

Thanks.
